pkg/kubelet/kubelet_pods.go

```diff
@@ -462,15 +471,19 @@ func nodeHostsFileContent(hostsFilePath string, hostAliases []v1.HostAlias) ([]b
 
 // managedHostsFileContent generates the content of the managed etc hosts based on Pod IPs and other
 // information.
-func managedHostsFileContent(hostIPs []string, hostName, hostDomainName string, hostAliases []v1.HostAlias) []byte {
+func managedHostsFileContent(hostIPs []string, hostName, hostDomainName string, hostAliases []v1.HostAlias, isIPv6 bool) []byte {
 	var buffer bytes.Buffer
 	buffer.WriteString(managedHostsHeader)
-	buffer.WriteString("127.0.0.1\tlocalhost\n")                      // ipv4 localhost
-	buffer.WriteString("::1\tlocalhost ip6-localhost ip6-loopback\n") // ipv6 localhost
-	buffer.WriteString("fe00::0\tip6-localnet\n")
-	buffer.WriteString("fe00::0\tip6-mcastprefix\n")
-	buffer.WriteString("fe00::1\tip6-allnodes\n")
-	buffer.WriteString("fe00::2\tip6-allrouters\n")
+	buffer.WriteString("127.0.0.1\tlocalhost\n") // ipv4 localhost
+
+	if isIPv6 {
```

Comment:
Should we also filter out any `hostAliases` set for IPv6? So one can set any host aliases and kubelet only applies IPv4 addresses if IPv6 are not supported?

Reply:
I don't think we should override user intent, this is just something we are prepopulating that we know will not be right

Reply:
> // we cannot do something completely correct,
> // so we can only do simple and basically correct things.

As the comments say, we can only do the basic right things :)